### ISSUE TITLE
RubyDNS 2.0.0 available

### ISSUE THREAD

**Issue by ioquatix** [code-mancers/invoker#185]
This update is a big improvement to stability, if you have time to try it out and report any issues that would be great.

It might also assist with the IPv6 issue, since that wasn't previously supported by event machine which is no longer used.

**Comment by iffyuva** [comment]
I will try it out locally next week and let you know @ioquatix . Thanks for new release 👍

**Comment by iffyuva** [comment]
hi @ioquatix , I tried new rc1 release gem, and am facing this issue

```
/Users/yuva/.rvm/gems/ruby-2.4.1/gems/async-dns-0.10.0/lib/async/dns/handler.rb:137:in `run': unknown keyword: listen (ArgumentError)
	from /Users/yuva/.rvm/gems/ruby-2.4.1/gems/async-dns-0.10.0/lib/async/dns/server.rb:112:in `block (2 levels) in run'
	from /Users/yuva/.rvm/gems/ruby-2.4.1/gems/async-dns-0.10.0/lib/async/dns/server.rb:111:in `each'
	from /Users/yuva/.rvm/gems/ruby-2.4.1/gems/async-dns-0.10.0/lib/async/dns/server.rb:111:in `block in run'
	from /Users/yuva/.rvm/gems/ruby-2.4.1/gems/async-0.12.0/lib/async/task.rb:70:in `block in initialize'
```

Any idea whats the issue? Thanks.

**Comment by ioquatix** [comment]
Do you mind sharing the code you are using to setup RubyDNS?

**Comment by iffyuva** [comment]
Its here https://github.com/code-mancers/invoker/blob/425a5d71d/lib/invoker/power/dns.rb @ioquatix

**Comment by ioquatix** [comment]
The problem is here: https://github.com/code-mancers/invoker/blob/425a5d71dd9fc19d7119d9886f6715d6b4c72305/lib/invoker/power/powerup.rb#L17

You need to remove `listen: DNS.server_ports`. You can either add it to the `new` or directly set it up in `DNS#initialize`.

Additionally, it won't work inside EventMachine, but it will work absolutely fine on it's own thread, and will be just as efficient since it's also using asynchronous io.

**Comment by ioquatix** [comment]
See https://github.com/socketry/async-dns/blob/master/lib/async/dns/server.rb#L39 for how to call `super`.

**Comment by iffyuva** [comment]
Thanks for pointers @ioquatix . I don't know much internals of invoker. I will figure out when I find time.

**Comment by ioquatix** [comment]
I've just released RubyDNS 2.0.0.

Please consider updating and let me know any feedback. Thanks.

The main user visible change between rc and 2.0.0 was that the `listen:` option was replaced by simply suppling as the first argument.
